File: app/src/main/java/com/fstyle/structure_android/screen/searchresult/SearchResultViewModel.java
package com.fstyle.structure_android.screen.searchresult;

import android.os.Bundle;
import com.fstyle.structure_android.screen.BaseViewModel;
import com.fstyle.structure_android.screen.userdetail.UserDetailActivity;
import com.fstyle.structure_android.utils.Constant;
import com.fstyle.structure_android.utils.navigator.Navigator;

/**
 * Created by daolq on 1/9/18.
 */

public class SearchResultViewModel extends BaseViewModel
        implements SearchResultAdapter.ItemClickListener {

    private SearchResultAdapter mAdapter;
    private Navigator mNavigator;

    public SearchResultViewModel(SearchResultAdapter adapter, Navigator navigator) {
        mAdapter = adapter;
        mNavigator = navigator;
        mAdapter.setItemClickListener(this);
    }

    @Override
    protected void onStart() {

    }

    @Override
    protected void onStop() {

    }

    @Override
    public void onItemClicked(Integer userId) {
        if (userId == null) {
            return;
        }
        Bundle bundle = new Bundle();
        bundle.putInt(Constant.ARGUMENT_USER_ID, userId);
        mNavigator.startActivity(UserDetailActivity.class, bundle);
    }

    public SearchResultAdapter getAdapter() {
        return mAdapter;
    }
}

File: app/src/main/java/com/fstyle/structure_android/screen/searchresult/SearchResultViewModel.kt
package com.fstyle.structure_android.screen.searchresult

import android.os.Bundle
import com.fstyle.structure_android.data.model.User
import com.fstyle.structure_android.screen.userdetail.UserDetailActivity
import com.fstyle.structure_android.utils.Constant
import com.fstyle.structure_android.utils.navigator.Navigator

/**
 * Created by le.quang.dao on 21/03/2017.
 */

class SearchResultViewModel(
    private val mPresenter: SearchResultContract.Presenter,
    val adapter: SearchResultAdapter,
    private val navigator: Navigator) : SearchResultContract.ViewModel, ItemUserClickListener {

  init {
    mPresenter.setViewModel(this)
    this.adapter.setItemClickListener(this)
  }

  override fun onStart() {
    mPresenter.onStart()
  }

  override fun onStop() {
    mPresenter.onStop()
  }

  override fun onItemUserClick(user: User) {
    val bundle: Bundle = Bundle()
    bundle.putString(Constant.ARGUMENT_USER_LOGIN, user.login)
    navigator.startActivity(UserDetailActivity::class.java, bundle)
  }
}
